State.h

```diff
@@ -0,0 +1,82 @@
+#ifndef STATE_H
+#define STATE_H
+
+#include "BreathData.h"
+
+/*
+ *  Parent class for all instantiated state classes
+ */
+class State {
+  public:
+    virtual State* update() = 0;           // perform maintenance, and perhaps transition to new state
```

Comment:
Why isn't the `enter()` method not also a virtual function in the base class?

Reply:
Now it is.